docs/src/main/asciidoc/spring-di.adoc

```diff
@@ -1,42 +1,116 @@
-////
-This guide is maintained in the main Quarkus repository
-and pull requests should be submitted there:
+//Only enable one of the following attributes at the same time
+// enable this attribute to show downstream markup and content, must be disabled for upstream doc builds
+//:downstream:
+// enable this attribute to show downstream markup and content, must be disabled for downstream doc builds
+:upstream:
+
+:doctype: book
+
+ifdef::downstream[]
+ifndef::context[]
+[id="proc-using-the-quarkus-extension-for-spring-di-api"]
+endif::[]
+ifdef::context[]
+[id="proc-using-the-quarkus-extension-for-spring-di-api_{context}"]
+endif::[]
+endif::[]
```

Comment:
I find using conditionals to conditionalize conditionals is kind of scary. When I look at this from a contributor's perspective it hurts my eyes. If the purpose is to insert `_{context}` into the IDs then maybe a solution might be to script that after pulling into the downstream repo.

That said I don't know that much about the approach for building these docs downstream. But it seems like a lot of the changes in this PR could be achieved with some grep, sed, awk magic and be less disruptive to community docs.

Reply:
@oraNod I agree that it looks horrible, yes. Can you show me how to do it with sed/awk/grep? I'd be more in favor adding such metadata using a script. Rather than having them all in the community version form the start. Even though it might not be a problem for Red Hat writer s to put up with this markup, community contributors might not like it, an I'd rather not force them to write it.

Reply:
cd docs/src/main/asciidoc
grep -r "^\[id=\"*"
sed -i 's|\[id="*[^"]*|&_{context}|g' *.adoc

One thing to note is that I did observe some single quotes `[id='foo']` instead of double quotes `[id="foo"]` so it would be ideal to standardize that first. For single quotes the result of that sed command is `[id='my-anchor']_{context}` 

I checked the templates, which use double quotes so probably go with that. Although I don't think it matters. For Infinispan we use single quotes.